pkg/cloud/services/compute/instance.go

```diff
@@ -31,21 +30,22 @@ import (
 	"github.com/gophercloud/gophercloud/openstack/networking/v2/networks"
 	"github.com/gophercloud/gophercloud/openstack/networking/v2/ports"
 	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
-	"github.com/gophercloud/utils/openstack/compute/v2/flavors"
 	"k8s.io/apimachinery/pkg/runtime"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	"sigs.k8s.io/cluster-api/util"
 
 	infrav1 "sigs.k8s.io/cluster-api-provider-openstack/api/v1beta1"
-	"sigs.k8s.io/cluster-api-provider-openstack/pkg/metrics"
 	"sigs.k8s.io/cluster-api-provider-openstack/pkg/record"
 	capoerrors "sigs.k8s.io/cluster-api-provider-openstack/pkg/utils/errors"
 )
 
+// Polling interval between retries during instance creation
+// N.B. This is overridden in unit tests.
+var retryIntervalInstanceStatus = 10 * time.Second
```

Comment:
Rather than using a global var, did you consider keeping this as a `const`, adding `retryInterval` as an argument to the function, and using the value of the const when the function is called? Then this value can be maintained as constant in the codebase but you can choose what to set it as in the tests.

Reply:
I've added https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1061/commits/75ba1a99d206156d0f0889edc14a301f1b8027d8 to see if I like this better than the global. Have a look and tell me what you think. I'm personally undecided on which version is nicer, so I'm very much interested in a second opinion!

I'll squash this commit if we decide it's better, so:

/hold

Reply:
I could be a bit paranoid about global variables, but I like passing the retry interval into the function just a little better. It has the drawback of adding a function argument which isn't absolutely essential, but I like that the dependency on the value is made clear in the function signature and that we maintain the original value as a const. What do you think? Maybe one day if more arguments are added to the function, we can tidy up by passing a config struct.

Reply:
Ok, I'll squash the new commit into the existing one.

Thanks for the reviews!